auth/services/util.go

```diff
@@ -1,22 +1,87 @@
 package services
 
 import (
+	"fmt"
 	ssi "github.com/nuts-foundation/go-did"
 	"github.com/nuts-foundation/go-did/did"
 	"github.com/nuts-foundation/nuts-node/vdr/types"
 	"time"
 )
 
-// ResolveEndpointURL finds the endpoint with the given type of the given holder and unmarshals it as single URL.
+// ResolveServiceURL finds the endpoint with the given type of the given holder referenced from the given service, and unmarshals it as single URL.
```

Comment:
I find it hard to wrap my head around this comment, especially the "referenced from the given service".

Probably because the name and signature of this function confuses me: it contains a `serviceType` and `endpointType` which are from DID spec perspective, the same thing. A `service` has a `type` property, and a `serviceEndpoint` which can be whatever is spec'd. I'd make the naming more explicit/elaborate;

`ResolveCompoundServiceURL(resolver, holder, compoundServiceType, serviceType, validAt)`

Reply:
Renamed the service and added some more lines of documentation.